modules/ossm-deploying-automatic-gateway-injection.adoc

```diff
@@ -6,66 +6,78 @@
 [id="ossm-deploying-automatic-gateway-injection_{context}"]
 = Deploying automatic gateway injection
 
-When deploying a gateway, you must opt-in to injection by adding an injection label or annotation to the gateway `deployment` object. The following example  `ConfigMap` object deploys a gateway with automatic injection.
+When deploying a gateway, you must opt-in to injection by adding an injection label or annotation to the gateway `deployment` object. The following example deploys a gateway.
 
 .Prerequisites
 
 * The namespace must be a member of the mesh by defining it in the `ServiceMeshMemberRoll` or by creating a `ServiceMeshMember` resource.
 
-.Example deployment with annotations
+.Procedure
+
+. Set a unique label for the Istio ingress gateway. This setting is required to ensure that the gateway can select the workload. This example uses ingressgateway as the name of the gateway.
++
 [source,yaml]
 ----
-  apiVersion: v1
-  kind: Service
-  metadata:
-    name: istio-ingressgateway
-    namespace: istio-ingress
-  spec:
-    type: LoadBalancer
-    selector:
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-ingress
+spec:
+  type: ClusterIP
+  selector:
+    istio: ingressgateway
+  ports:
+  - name: http
+    port: 80
+    targetPort: 8080
+  - name: https
+    port: 443
+    targetPort: 8443
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-ingress
+spec:
+  selector:
+    matchLabels:
       istio: ingressgateway
-    ports:
-    - port: 80
-      name: http
-    - port: 443
-      name: https
-  ---
-  apiVersion: apps/v1
-  kind: Deployment
-  metadata:
-    name: istio-ingressgateway
-    namespace: istio-ingress
-  spec:
-    selector:
-      matchLabels:
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
         istio: ingressgateway
-    template:
-      metadata:
-        annotations:
-          inject.istio.io/templates: gateway <1>
-        labels:
-          istio: ingressgateway <2>
-          sidecar.istio.io/inject: "true" <3>
-      spec:
-        containers:
-        - name: istio-proxy
-          image: auto <4>
-  ---
-  apiVersion: rbac.authorization.k8s.io/v1 <5>
-  kind: Role
-  metadata:
-    name: istio-ingressgateway-sds
-    namespace: istio-ingress
-  rules:
-    - apiGroups: [""]
-      resources: ["secrets"]
-      verbs: ["get", "watch", "list"]
-  ---
-  apiVersion: rbac.authorization.k8s.io/v1
-  kind: RoleBinding
-  metadata:
-   name: istio-ingressgateway-sds
-   namespace: istio-ingress
+        sidecar.istio.io/inject: "true" <1>
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto <2>
+----
+<1> Enable gateway injection by setting the `sidecar.istio.io/inject` field to `"true"`. When connecting to a revisioned control plane, replace with `istio.io/rev: revision-name`.
```

Comment:
`When connecting to a revisioned control plane, replace with `istio.io/rev: revision-name`.`

It was not clear about this step. Please add more info to understand in a better way.

Reply:
Maybe let's remove this as it just adds confusion on a subject which isn't necessarily what we're talking about here (deploying a gateway). However, for reference see https://istio.io/latest/blog/2021/revision-tags albeit I'm not sure if we in OSSM even support this?